Elemental/Components/AeTable/AePaginatedTable.razor

```diff
@@ -86,18 +86,26 @@
 
     protected override async Task OnParametersSetAsync()
     {
-        LoadPageItems(true);
         Total = Rows.Count();
+        int PageCount = (int)Math.Ceiling(Total / (decimal)Math.Max(1, PageSize));
+        Page = (Page > PageCount - 1) ? PageCount - 1 : Page;
```

Comment:
What's going on here?

Reply:
Ah, sorry for the late reply!
This line here checks if the current Page is greater than PageCount (out of bounds) and sets Page to PageCount if Page is out of bounds.
This happens when you switch the dataset because the currently displayed page will update, but the value Page isn't modified. In this case, if Page > PageCount then the displayed page will just show an empty page with no content.
If there's a better fix, then we can use that instead!

Reply:
Does it throw an exception if the Page is properly set when the dataset is changed?

I'm thinking we just check for an out-of-bounds `[0, count)` and set to `0` if it fails.

Reply:
I created a CheckOutOfBounds() function which serves the same purpose as well as checking if Page < 0.
It doesn't throw any exceptions, at least with the tests I did.